### ISSUE TITLE
Add RamWorks /r option to UI

### ISSUE THREAD

**Issue by tomcw** [AppleWin/AppleWin#27]
**BerliOS Feature Request #2222**

**Date:** 2006-May-19 14:48
**Submitted By:** poake
**Category:** none
**Assigned To:** tomch
**Priority:** 5
# 

Date: 2006-May-19 14:48
Sender: poake
Logged In: YES 
user_id=26305
Browser: Mozilla/5.0 (Windows; U; Windows NT 5.1; en-US; rv:1.8.0.3) Gecko/20060426
Firefox/1.5.0.3

I'd like to see the ability to add memory to the
emulator added to the config screen with a display
showing what it is currently set at.

Thanks :)

**Comment by Michaelangel007** [comment]
I've been thinking about adding Saturn 128K (language card) support ... we probably should have a new config tab called `Memory`.

```
( ) RamWorks
    Number of Banks:       _127
    Total extended memory: 8128 KB

( ) Saturn 64K
( ) Saturn 128K
```

Note: The user enters in the number of banks -- we calculate and display the total available memory in KB.

We also need to make sure extended memory is saved/loaded from the save state.

**Comment by Spiny-Norman** [comment]
The Config UI is currently a little tricky: adding a 6th tab to the Config (depending on the tab's title length!) may or may not split the current single tab row into 2 rows. And I really don't like 2 tabs row (as a tab on the lower row gets "promoted" to the upper row when selected), which I find confusing.

Anyway, I'm sure we can fiddle with each of the 6 tab titles to accommodate 6 tabs.

**Comment by tomcw** [comment]
The Config UI is currently a little tricky: adding a 6th tab to the Config (depending on the tab's title length!) may or may not split the current single tab row into 2 rows. And I really don't like 2 tabs row (as a tab on the lower row gets "promoted" to the upper row when selected), which I find confusing.

Anyway, I'm sure we can fiddle with each of the 6 tab titles to accommodate 6 tabs.

**Comment by Michaelangel007** [comment]
Yeah, I personally hate two line tabs for the same reason you mentioned Tom.

If we make that dialog wider can we force a single row for tabs?

**Comment by tomcw** [comment]
Yes, making the dialog wider or the tab name shorter will ensure just a single row of tabs.

**Comment by tomcw** [comment]
What slot does the Saturn go in? (Presumably any?)
Can it go into slot-0 for the Apple II / II+?
Can it go into aux for the (Enhanced) //e?

**Comment by Michaelangel007** [comment]
My understanding is that, yes, the Saturn can go in any Slot.

**Comment by Michaelangel007** [comment]
If we don't support _only_ 48K then we should remove the first option:

` ( ) None`

We probably should rename the second option:

`( ) Apple 16K Language Card`

That way everything stays in nice alphabetical order. 👍

**Comment by tomcw** [comment]
In #408 you said that:
>The Saturn card is a drop-in replacement for Apple's 16K Language Card

So presumably it doesn't really make sense to put it into a //e.
Therefore should we only support for II and II+ models (and clones of the II/II+)?
And so just hardcode it to slot-0 only?

**Comment by Michaelangel007** [comment]
> So presumably it doesn't really make sense to put it into a //e.

From a _practical_ point of view, yeah, since you already have Aux mem, but knowing Apple 2 owners ... _"just because you can, means we can"_ :)

> And so just hardcode it to slot-0 only?

I'm fine with that.  I'm going to keep punting on "config any card in any slot" for as long as possible. ;-)

**Comment by Michaelangel007** [comment]
> Therefore should we only support for II and II+ models (and clones of the II/II+)?

I don't think there is any advantage to limiting it to just the II, and II+.  While technically all we just need to do is wrap:

```
if (g_uSaturnTotalBanks)
```

With one the APPLE2 macros, you know Murphy's Law is going to pop up: We're going to get a AppleWin //e user trying to use the Saturn card. ;-)

**Comment by tomcw** [comment]
Yes, I agree. And it's simpler if we allow support for all A2 models (in AppleWin).
